src/main/java/com/geespring/marketapi/service/CartService.java

```diff
@@ -0,0 +1,76 @@
+package com.geespring.marketapi.service;
+
+import com.geespring.marketapi.dto.cart.CartProductDTO;
+import com.geespring.marketapi.model.Cart;
+import com.geespring.marketapi.repository.CartRepository;
+import com.geespring.marketapi.util.exceptions.DataNotFoundException;
+import com.google.common.collect.ImmutableList;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class CartService {
+
+    //TODO replace static id '1' with session user id
+
+    /**
+     * Represents the Cart Repository
+     */
+    @Autowired
+    private final CartRepository repository;
+
+    /**
+     * Fetches a list of all products in the users cart
+     * @return the list of products
+     */
+    public List<Cart> getCartProducts(final Long userId) {
+        return ImmutableList.copyOf(repository.findCartsByUserId(userId));
+    }
+
+    /**
+     * Adds the specified product id into the users cart
+     * @param product
+     */
+    public Cart add(final Long userId, final CartProductDTO product) {
```

Comment:
Can we rename this to `addProduct` for more clarity?

Reply:
Also, this will simply update quantity for an existing cart - not update the actual product _and_ quantity inside it. Somewhat misleading to other developers on the team. Let's remove the functionality to update from this method in particular, because methods should typically do one thing only.